x-pack/plugins/observability_solution/observability_ai_assistant/public/hooks/use_user_preferred_language.ts

```diff
@@ -0,0 +1,218 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { useCallback } from 'react';
+import useLocalStorage from 'react-use/lib/useLocalStorage';
+
+const USE_KIBANA_LOCALE_SETTING = 'Use Kibana locale setting';
+const USE_BROWSER_LANGUAGE_SETTING = 'Use browser language setting';
```

Comment:
Should these be `i18n`-ised? And why the they named as constants (upper-case)? This indicates to me that they are used as keys and should not be changed in the future. Is that the case? (pun intended :p)

Reply:
I use them to compare with the selected option to decide how I should resolve the language, and see my comment below about why I think I cannot translate them.